components/server/src/user/user-service.ts

```diff
@@ -205,6 +206,28 @@ export class UserService {
         return false;
     }
 
+    /**
+     * Identifies the team to which a workspace instance's running time should be attributed to
+     * (e.g. for usage analytics or billing purposes).
+     * If no specific team is identified, the usage will be attributed to the user instead (default).
+     *
+     * @param user
+     * @param projectId
+     */
+    async getWorkspaceUsageAttributionTeamId(user: User, projectId?: string): Promise<string | undefined> {
```

Comment:
Placing this method in `UserService` is debatable. For example, this logic could be in-lined in `workspace-starter.ts` (which already has `projectDb` injected).

My reason for placing it here was to allow an extension point where ee/UserService could detect if payment is enabled, and use a different logic (i.e. attribute all usage to a "cost center" team, instead of the project's team).

Reply:
Not strongly opinionated here either but your reasoning here makes sense; the subject of cost center vs team has already come up in discussions.